sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -502,8 +503,18 @@ final class DataFrameWriter private[sql](df: DataFrame) {
 
   private def assertNotBucketed(): Unit = {
     if (numBuckets.isDefined || sortColumnNames.isDefined) {
+      val ourMethodName = "assertNotBucketed"
+      val callerMethodName = Thread.currentThread.getStackTrace()
```

Comment:
I see similar stuff done here to get the caller name : https://github.com/apache/spark/blob/master/streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala#L262

This logic could be moved to a util method. Utils class seems to have some method regarding getting callsite : https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/Utils.scala#L1331